api/v1alpha1/velero_types.go

```diff
@@ -54,47 +54,70 @@ const CSIPluginImageKey UnsupportedImageKey = "csiPluginImageFqin"
-	DefaultVeleroPlugins []DefaultPlugin `json:"defaultVeleroPlugins,omitempty"`
+	ResourceAllocations corev1.ResourceRequirements `json:"resourceAllocations,omitempty"`
+	// DefaultPlugins defines the list of default plugins (aws, gcp, azure, openshift, csi) to be installed with Velero
+	DefaultPlugins []DefaultPlugin `json:"defaultPlugins,omitempty"`
 	// CustomVeleroPlugins defines the custom plugin to be installed with Velero
 	// +optional
-	CustomVeleroPlugins []CustomPlugin `json:"customVeleroPlugins,omitempty"`
+	CustomPlugins []CustomPlugin `json:"customPlugins,omitempty"`
+	// RestoreResourceVersionPriority represents a configmap that will be created if defined for use in conjunction with `EnableAPIGroupVersions` feature flag
+	// Defining this field automatically add EnableAPIGroupVersions to the velero server feature flag
+	// +optional
+	RestoreResourcesVersionPriority string `json:"restoreResourcesVersionPriority,omitempty"`
+	// If you need to install Velero without a default backup storage location NoDefaultBackupLocation flag is required for confirmation
+	// +optional
+	NoDefaultBackupLocation bool `json:"noDefaultBackupLocation,omitempty"`
+}
 
-	// EnableRestic is a boolean to specify if restic daemonset instance should be created or not
+type ResticConfig struct {
+	// SupplementalGroups defines the linux groups to be applied to the Restic Pod
 	// +optional
-	EnableRestic *bool `json:"enableRestic,omitempty"`
-	// ResticSupplementalGroups defines the linux groups to be applied to the Restic Pod
+	SupplementalGroups []int64 `json:"supplementalGroups,omitempty"`
+	// NodeSelector defines the nodeSelector to be supplied to Restic podSpec
 	// +optional
-	ResticSupplementalGroups []int64 `json:"resticSupplementalGroups,omitempty"`
-	// ResticNodeSelector defines the nodeSelector to be supplied to Restic podSpec
+	NodeSelector map[string]string `json:"nodeSelector,omitempty"`
+	// Tolerations defines the list of tolerations to be applied to Restic daemonset
 	// +optional
-	ResticNodeSelector map[string]string `json:"resticNodeSelector,omitempty"`
-	// ResticTolerations defines the list of tolerations to be applied to Restic daemonset
+	Tolerations []corev1.Toleration `json:"tolerations,omitempty"`
+	// ResourceAllocations defines the CPU and Memory resource allocations for the restic Pod
 	// +optional
-	ResticTolerations []corev1.Toleration `json:"resticTolerations,omitempty"`
-	// ResticResourceAllocations defines the CPU and Memory resource allocations for the restic Pod
+	ResourceAllocations corev1.ResourceRequirements `json:"resourceAllocations,omitempty"`
+	// Timeout defines the Restic timeout, default value is 1h
 	// +optional
-	ResticResourceAllocations corev1.ResourceRequirements `json:"resticResourceAllocations,omitempty"`
-	// ResticTimeout defines the Restic timeout, default value is 1h
+	Timeout string `json:"timeout,omitempty"`
+}
+
+type ServerConfig struct {
+	*VeleroConfig
```

Comment:
I think this might have issues,

```ServerConfig {
   tolerations: .......
  }
```

Will be undefined for either the velero config or restic config. I think that we should have NamedFields for these.

Reply:
Great point, thanks Shawn